File: java/src/test/java/challenges/aoc2024/day3/model/MultiplicationInstructionTest.java
package challenges.aoc2024.day3.model;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class MultiplicationInstructionTest {

    @Test
    void shouldBuildMultiplicationInstructionFromString() {
        // When
        final MultiplicationInstruction instruction = MultiplicationInstruction.of("mul(123,99)");

        // Then
        assertEquals(ImmutableMultiplicationInstruction.class, instruction.getClass());
        assertEquals(123, instruction.getFirstValue());
        assertEquals(99, instruction.getSecondValue());
    }

    @Test
    void shouldBuildMultiplicationInstructionFromValues() {
        // When
        final MultiplicationInstruction instruction = MultiplicationInstruction.of(123, 99);

        // Then
        assertEquals(ImmutableMultiplicationInstruction.class, instruction.getClass());
        assertEquals(123, instruction.getFirstValue());
        assertEquals(99, instruction.getSecondValue());
    }

}

File: scala/src/test/scala/challenges/aoc2024/day3/model/MultiplicationInstructionTest.scala
package challenges.aoc2024.day3.model

import org.scalatest.funsuite.AnyFunSuite

class MultiplicationInstructionTest extends AnyFunSuite {

  test("should build MultiplicationInstruction from string") {
    // When
    val instruction: MultiplicationInstruction = MultiplicationInstruction.of("mul(123,99)")

    // Then
    assert(instruction.isInstanceOf[MultiplicationInstruction])
    assert(123 == instruction.firstValue)
    assert(99 == instruction.secondValue)
  }

  test("should build MultiplicationInstruction from values") {
    // When
    val instruction: MultiplicationInstruction = MultiplicationInstruction(123, 99)

    // Then
    assert(instruction.isInstanceOf[MultiplicationInstruction])
    assert(123 == instruction.firstValue)
    assert(99 == instruction.secondValue)
  }
}
